### PR TITLE
Moving from travis.ci to github-actions

### PR DESCRIPTION
**Pull request by eckerpatrick**
Since the travis CI stopped working its a good time to transfer the CI to github actions

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build and Test
+
+on: [push, pull_request]
+
+env:
+  ROOT_VERSION: root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
```
Comment (GiacomoXT) [inline .github/workflows/cmake.yml:10]: I suggest to have an additional build running on Ubuntu 22 as well.

Maybe you want to test also a newer version of ROOT? Like `6.28/04`.

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        source root/bin/thisroot.sh
+        cd ${{github.workspace}}/.. && mkdir build && cd build
+        cmake ../GenFit
+        make
+    
+    - name: Run Tests
+      run: |
+        source root/bin/thisroot.sh
+        cd ${{github.workspace}}/../build
+        ctest
```
Comment (GiacomoXT) [inline .github/workflows/cmake.yml:41]: I see that the travis build was using `ctest --output-on-failure`. Should you use the same here?

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (eckerpatrick) [reply .github/workflows/cmake.yml:10]: So yes I think it's a good idea to test as much as possible. Nevertheless, we have 2,000 min available per month (as far as I understood the pricing). Given that currently, not that much development is going on this should be fine, but something to keep in mind.

Comment (GiacomoXT) [reply .github/workflows/cmake.yml:10]: (Can we register a "private" runner, in case?)

Comment (eckerpatrick) [reply .github/workflows/cmake.yml:10]: Yes, this is possible.

File: .github/workflows/cmake.yml (hunk 2 above)
Comment (eckerpatrick) [reply .github/workflows/cmake.yml:41]: Yes, that makes sense

**Comment by eckerpatrick** [comment]
I will merge this PR now to have some pipeline back in place.
Additional improvements to this pipeline can then be done in an additional PR.

**Pull request merged**